File: Tau/src/core/Synth/Editor/InstFragment.cpp
#include"InstFragment.h"
#include"Midi/MidiEvent.h"

namespace tau
{
	InstFragment::InstFragment()
	{

	}

	InstFragment::~InstFragment()
	{
		LinkedListNode<MidiEvent*>* node = midiEvents.GetHeadNode();
		for (; node; node = node->next)
		{
			DEL(node->elem);
		}

		midiEvents.Release();
	}

	void InstFragment::Clear()
	{
		eventOffsetNode = midiEvents.GetHeadNode();
	}

	void InstFragment::AddMidiEvents(LinkedList<MidiEvent*>& midiEvents_)
	{
		midiEvents.Merge(midiEvents_);
	}


}

File: Tau/andstudio/Tau/app/src/main/java/cymheart/tau/editor/InstFragment.java
package cymheart.tau.editor;

import java.util.LinkedList;
import java.util.List;
import java.util.ListIterator;

import cymheart.tau.midi.MidiEvent;
import cymheart.tau.utils.ScLinkedList;
import cymheart.tau.utils.ScLinkedListNode;

public class InstFragment {

    protected int startTick = 0;
    protected double startSec = 0;
    protected double endSec = 0;

    protected Track track = null;
    protected int branchIdx = 0;

    //
    protected MidiEvent[] _ndkMidiEvent;
    protected ScLinkedList<MidiEvent> midiEvents = new ScLinkedList<>();
    protected ScLinkedListNode<MidiEvent> eventOffsetNode;
    protected ScLinkedListNode<MidiEvent> eventFirstNode;

    public ScLinkedList<MidiEvent> GetMidiEvents() {
        return midiEvents;
    }

    void Clear()
    {
        eventOffsetNode = eventFirstNode = midiEvents.GetHeadNode();
    }

}
